blocks/library/embed/index.js

```diff
@@ -26,7 +27,27 @@ import BlockAlignmentToolbar from '../../block-alignment-toolbar';
 // These embeds do not work in sandboxes
 const HOSTS_NO_PREVIEWS = [ 'facebook.com' ];
 
-function getEmbedBlockSettings( { title, icon, category = 'embed', transforms, keywords = [] } ) {
+// Caches the embed API calls, so if blocks get transformed, or deleted and added again, we don't spam the API.
+const wpEmbedAPI = memoize( ( url ) => wp.apiRequest( { path: `/oembed/1.0/proxy?${ stringify( { url } ) }` } ) );
+
+const matchesPatterns = ( url, patterns ) => {
+	return patterns.some( ( pattern ) => {
+		return url.match( new RegExp( pattern, 'i' ) );
```

Comment:
I don't think we need to construct this from a string any longer, just define the original values as RegExp in the first place.

e.g.

**Before:**

```js
patterns: [ '^https?:\/\/(www\.)?speakerdeck\.com\/.+' ],
```

**After:**

```js
patterns: [ /^https?:\/\/(www\.)?speakerdeck\.com\/.+/i ],
```

Reply:
Done!